util/util_test.go

```diff
@@ -903,5 +905,74 @@ func TestPathSliceSort(t *testing.T) {
 				t.Errorf("path sort failed: wanted: %s got: %s", tt.out[i], tt.in[i])
 			}
 		}
+
+func TestCopyFs1(t *testing.T) {
+	src := afero.NewMemMapFs()
+	dst := afero.NewMemMapFs()
+
+	fooData := []byte("foo")
+	barData := []byte("bar")
+	bazData := []byte("baz")
+
+	src.MkdirAll("/foo/bar/baz/", 0700)
+	afero.WriteFile(src, "/foo/foo.txt", fooData, 0600)
+	afero.WriteFile(src, "/foo/bar/bar.txt", barData, 0600)
+	afero.WriteFile(src, "/foo/bar/baz/baz.txt", bazData, 0600)
+	CopyFs(src, dst, "", "", false)
+
+	contains, _ := afero.FileContainsBytes(dst, "/foo/foo.txt", fooData)
+	if !contains {
+		t.Errorf("Expected /foo/foo.txt to contain %s", fooData)
+	}
+
+	contains, _ = afero.FileContainsBytes(dst, "/foo/bar/bar.txt", barData)
+	if !contains {
+		t.Errorf("Expected /foo/bar/bar.txt to contain %s", barData)
+	}
+
+	contains, _ = afero.FileContainsBytes(dst, "/foo/bar/baz/baz.txt", bazData)
+	if !contains {
+		t.Errorf("Expected /foo/bar/baz/baz.txt to contain %s", bazData)
+	}
+}
+
+func TestCopyFs2(t *testing.T) {
```

Comment:
Since TestCopyFs2, 3 and 4 are the same, but with different inputs and outputs, you might want to consider combining them into a table driven test and loop through the parameters, like in https://github.com/purpleidea/mgmt/blob/bd80ced9b238016271c55d93e052eaf25344f1c9/util/util_test.go#L861.

Not strictly necessary, but useful if we want to quickly add more test cases down the line.

Reply:
Sure. This is a good idea.